t4.py

```diff
@@ -1,125 +1,121 @@
 import requests
 import json
 import os
-
-# 1. Hardcoded sensitive credentials and secrets (BAD PRACTICE)
-DATABASE_USERNAME = "admin"  # Hardcoded username (BAD)
-DATABASE_PASSWORD = "password123"  # Hardcoded password (BAD)
```

Comment:
Hardcoding sensitive information such as database credentials, API keys, and secret keys directly in the source code is a severe security risk. This practice exposes sensitive data to anyone who has access to the codebase, increasing the risk of data breaches. It's recommended to use environment variables or secure vaults for managing sensitive information.

**Label**: security

**Code suggestion**

```python
# Use environment variables or a secure vault
DATABASE_USERNAME = os.getenv('DATABASE_USERNAME')
DATABASE_PASSWORD = os.getenv('DATABASE_PASSWORD')
API_KEY = os.getenv('API_KEY')
SECRET_KEY = os.getenv('SECRET_KEY')
```